src/main/java/uk/gov/gchq/palisade/util/AbstractResourceBuilder.java

```diff
@@ -79,15 +81,38 @@ public static Resource create(final String uriString) {
      * @param uri the uri of the resource you want built
      * @return a newly created resource with the id of the uri.
      */
+    @SuppressWarnings("java:S1075") //Suppress hardcoded path separator code smell
     public Resource buildNormal(final URI uri) {
-        try {
-            URI normal = UriBuilder.create(uri)
+        var absoluteResourceId = uri;
+
+        if (!uri.getSchemeSpecificPart().startsWith("/")) {
+            var localResource = new File(uri.getSchemeSpecificPart());
+            String path;
+            try {
+                path = localResource.getCanonicalPath();
+            } catch (IOException e) {
+                LOGGER.warn("Unable to get the Canonical path value", e);
+                path = localResource.getAbsolutePath();
```

Comment:
From what I understand about the distinction of a Canonical and Absolute path, I would have expected that we would use only the Canonical or throw an exception.  The Absolute can be ambiguous so using it could create a an issue.  Also the construction of the path looks like a method in its own right.  One small benefit would be that it would be easier to test.